tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/kg/ResourcesSpec.scala

```diff
@@ -69,95 +67,158 @@ class ResourcesSpec extends BaseIntegrationSpec {
 
   override def beforeAll(): Unit = {
     super.beforeAll()
-    (createProjects(Rick, orgId, projId1, projId2) >>
-      aclDsl.addPermission(s"/$project1", Morty, Resources.Read)).accepted
-    ()
+    val setup = for {
+      _                    <- createOrg(Rick, orgId)
+      forcedSchemaPayload   = ProjectPayload.generate(projectForcedSchema, config, enforceSchema = true)
+      _                    <- adminDsl.createProject(orgId, projForcedSchemaId, forcedSchemaPayload, Rick)
+      optionalSchemaPayload = ProjectPayload.generate(projectOptionalSchema, config, enforceSchema = false)
+      _                    <- adminDsl.createProject(orgId, projOptionalSchemaId, optionalSchemaPayload, Rick)
+      _                    <- aclDsl.addPermission(s"/$projectForcedSchema", Morty, Resources.Read)
+    } yield ()
+    setup.accepted
   }
 
-  "adding schema" should {
+  "Adding schema" should {
     "create a schema" in {
       val schemaPayload = SchemaPayload.loadSimple().accepted
-
-      deltaClient.put[Json](s"/schemas/$project1/test-schema", schemaPayload, Rick) { (_, response) =>
-        response.status shouldEqual StatusCodes.Created
-      }
+      deltaClient.put[Json](s"/schemas/$projectForcedSchema/test-schema", schemaPayload, Rick) { expectCreated }
     }
 
     "creating a schema with property shape" in {
       val schemaPayload = jsonContentOf("kg/schemas/simple-schema-prop-shape.json")
-
-      deltaClient.post[Json](s"/schemas/$project1", schemaPayload, Rick) { (_, response) =>
-        response.status shouldEqual StatusCodes.Created
-      }
+      deltaClient.post[Json](s"/schemas/$projectForcedSchema", schemaPayload, Rick) { expectCreated }
     }
 
     "creating a schema that imports the property shape schema" in {
       val schemaPayload = jsonContentOf("kg/schemas/simple-schema-imports.json")
-
-      eventually {
-        deltaClient.post[Json](s"/schemas/$project1", schemaPayload, Rick) { (_, response) =>
-          response.status shouldEqual StatusCodes.Created
-        }
-      }
+      deltaClient.post[Json](s"/schemas/$projectForcedSchema", schemaPayload, Rick) { expectCreated }
     }
   }
 
-  "creating a resource" should {
+  "Creating a resource" should {
```

Comment:
Most tests related to creating a resource which were scattered in the file are now grouped here